### PR TITLE
WT-7384 Fix an assert fire when inserting to the history store

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/history/hs_rec.c [hunk 1]
```diff
@@ -142,13 +142,15 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
               &upd_type_full_diag, existing_val));
             WT_ERR(__wt_compare(session, NULL, existing_val, hs_value, &cmp));
             /*
-             * We shouldn't be inserting the same value again for the key unless coming from a
-             * different transaction. If the updates are from the same transaction, the start
-             * timestamp for each update should be different.
+             * We shouldn't be inserting the same value again for the key unless it comes from a
+             * different transaction or it is previously deleted by rollback to stable or prepared
```
Comment (hbokhari) [inline src/history/hs_rec.c:146]: it is previously deleted -> it was previously deleted

**Review by hbokhari — APPROVED** [review]
Requesting a minor fix but LGTM otherwise.

File: src/history/hs_rec.c (hunk 1 above)
Comment (bvpvamsikrishna) [reply src/history/hs_rec.c:146]: I prefer the comment to explain the assertion. Something like " Same value should not be inserted again unless 1. previous entry is already deleted(i.e. either stop timestamp is globally visible or transaction is globally visible), 2. From a different transaction 3. with a different timestamp if from the same transaction".

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm,  improve the comment above the assertion.

**Pull request merged**